lib/src/data/repository/genre_repository_impl.dart

```diff
@@ -1,39 +1,40 @@
-import 'dart:convert';
 import 'dart:core';
-import 'package:flutter/services.dart';
+import '../../core/util/api_constants.dart';
 import '../../domain/entity/genre.dart';
 import '../../domain/repository/i_repository.dart';
+import '../datasource/remote/i_api_service.dart';
+import '../../core/util/data_state.dart';
+import '../model/genre_model.dart';
 
-class GenreRepositoryImpl implements IRepository<List<Genre>> {
-  @override
-  Future<List<Genre>> getData() async {
-    String jsonString =
-        await rootBundle.loadString('assets/jsonfile/genres.json');
-    Map<String, dynamic> jsonData = jsonDecode(jsonString);
-    List<Genre> genresList = [];
+class GenreRepositoryImpl implements IRepository<DataState<List<Genre>>> {
+  IApiService genresService;
 
-    for (var genre in jsonData['genres']) {
-      genresList.add(
-        Genre(
-          id: genre['id'],
-          name: genre['name'],
-        ),
-      );
-    }
+  GenreRepositoryImpl({
+    required this.genresService,
+  });
 
-    return genresList;
-  }
+  @override
+  Future<DataState<List<Genre>>> getData() async {
+    DataState data = await genresService.fetchDataFromApi();
+    if (data is DataSuccess) {
+      List<GenreModel> genreModel = data.data;
 
-  List<String> getRelatedGenres(
-    List<Genre> genres,
-    List<num> movieGenresIds,
-  ) {
-    List<String> movieRelatedGenres = [];
-    movieRelatedGenres = genres
-        .where((genre) => movieGenresIds.contains(genre.id))
-        .map((genre) => genre.name)
-        .toList();
+      if (genreModel.isNotEmpty) {
+        List<Genre> result = [];
+        for (GenreModel genre in genreModel) {
+          result.add(genre);
+        }
```

Comment:
It'd be better using `genreModel.map(...)`

Reply:
i used polymorphism instead, because GenreModel extends from Genre, 
```
    if (data is DataSuccess) {
      List<GenreModel> genreModel = data.data;

      if (genreModel.isNotEmpty) {
        return DataSuccess<List<Genre>>(
          genreModel,
        );
      }
```
is this okay?

Reply:
Even better!